crates/libcontainer/src/process/fork.rs

```diff
@@ -187,7 +185,7 @@ fn clone(cb: CloneCb, flags: u64, exit_signal: Option<u64>) -> Result<Pid, Clone
 
     // Since the child stack for clone grows downward, we need to pass in
     // the top of the stack address.
-    let child_stack_top = unsafe { child_stack.add(default_stack_size) };
+    let child_stack_top = unsafe { child_stack.as_ptr().add(default_stack_size) };
```

Comment:
Why is `as_ptr` needs to be added, when the return type of (original) `mmap` and (new) `mmap_anonymous` is same of `Result<NonNull<c_void>>` ?

Reply:
> Why is `as_ptr` needs to be added, when the return type of (original) `mmap` and (new) `mmap_anonymous` is same of `Result<NonNull<c_void>>` ?

Perhaps because with nix version "0.27.1", the mmap function returns a `Result<*mut c_void>`.

Reply:
oh yes, I was comparing both from 0.28.0, my bad. Thanks!